.github/workflows/test.yml

```diff
@@ -0,0 +1,89 @@
+name: Test
+on: [push, pull_request]
+jobs:
+  test:
+    needs: code-analysis
+    strategy:
+      fail-fast: false
```

Comment:
what is rationale for this?
fast fails will help to get feedback faster and save our minutes and developer can run extensive tests locally.